### PR TITLE
SWE: calculate phase space density

### LINKED ISSUE #884: Convert counts to phase space density
Calculate phase space density fv in units of s^3/cm^6

        fv(v,theta,phi) = 2*C(E,theta,phi)/(G*v^4*tau)

        C = corrected counts
        E = electron energy, in eV (result from caluclate_particle_energy())
        v = electron speed, computed from energy, in cm/s
            E = 0.5 * m * v^2
            where m is mass of electron. TODO: Ruth will send me conversion factors.
            do basic algebra to get v = sqrt(2 * E / m). E is calculated from this #883 
        tau = sampling time, in s. It's the ACQ_DURATION from packet. Use first
            quarter cycle's value.
        G = geometric factor, in (cm^2 * ster). 7 cems value. ster is steradian
            and is an angle.

### PR DESCRIPTION
**Pull request by tech3371**
# Change Summary
closes #884 
## Overview
Function to calculate SWE L2 phase space density. I used lot of ideas Greg suggested before leave.

## Updated Files
- imap_processing/swe/l1b/swe_l1b_science.py
   - update caused from refactor code
- imap_processing/swe/l2/swe_l2.py
   - main changes to calculate phase space density
- imap_processing/swe/utils/swe_utils.py
   - refactor way of reading in lookup table

## Testing
- imap_processing/tests/swe/test_swe_l2.py
   - added test for new function

### CODE REVIEW

File: imap_processing/swe/l2/swe_l2.py [hunk 1]
```diff
@@ -1,33 +1,99 @@
...
+import numpy as np
 import numpy.typing as npt
+import xarray as xr
 
-from imap_processing.swe.l1b.swe_l1b_science import read_lookup_table
+from imap_processing.swe.utils.swe_utils import read_lookup_table
 
-# TODO: add this to instrument status summary
+# TODO: add these to instrument status summary
 ENERGY_CONVERSION_FACTOR = 4.75
```
Comment (maxinelasp) [inline imap_processing/swe/l2/swe_l2.py:10]: Should these constants be in a swe constants.py type file? Fine if they're only used here, just wanted to ask

**Review by maxinelasp — APPROVED** [review]
LGTM, although I'm not super familiar with the rest of the code base. I appreciate the use of dataframes rather than dicts.

File: imap_processing/swe/l2/swe_l2.py (hunk 1 above)
Comment (tech3371) [reply imap_processing/swe/l2/swe_l2.py:10]: This is a temporary storage. I image these going into instrument status summary. Right now, it's only used here. I will keep `constants.py` idea in mind if we use this in another place later on.

**Comment by tech3371** [comment]
> LGTM, although I'm not super familiar with the rest of the code base. I appreciate the use of dataframes rather than dicts.

Thank you for quick review. I know you just go back.

dataframes ideas credit goes to Greg. I was not happy with what I had before and asked Greg for ideas. He was the one who suggested getting everything in same shape to simplify mathematical operations.

**Pull request merged**